flink-core/src/main/java/org/apache/flink/util/concurrent/DirectExecutorService.java

```diff
@@ -28,16 +28,17 @@
 import java.util.concurrent.Callable;
 import java.util.concurrent.CancellationException;
 import java.util.concurrent.ExecutionException;
-import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
 
 /**
- * The direct executor service directly executes the runnables and the callables in the calling
- * thread.
+ * The direct scheduled executor service directly executes the runnables and the callables in the
+ * calling thread.
  */
-class DirectExecutorService implements ExecutorService {
+class DirectExecutorService implements ScheduledExecutorService {
```

Comment:
I'm not sure about changing `DirectExecutorService`. The only reason I see for this change is that some shuffle test cases are using `Executors#newDirectExecutorService`. This also indicates that the code path for scheduling executions are never covered. Alternatively, we can make those tests to use a real scheduled executor service.

Reply:
Good suggestion, changed those tests to use a real scheduled executor service and revert the changing of `DirectExecutorService `.